sdks/python/apache_beam/transforms/core.py

```diff
@@ -189,6 +276,54 @@ def finish_bundle(self):
     """
     pass
 
+  def new_tracker(self, restriction):
```

Comment:
<!--thread_id:cc_141417252_t; commit:95a209dccc3398be93d21bd9d926a886343b9617; resolved:0-->
<!--section:context-quote-->
> **robertwb** wrote:
> The advantage is that we can make these many methods abstract such that the user won't forget any of them when they need them.

<!--section:body-->
I'm not 100% sure of this one. Seems like you are suggesting to ask users to directly implement the MixIn (say SplittableMixIn). But this means that every DoFn author will have to implement two classes due to the need to make it easy for the SDK to perform validation. I think it might be better to make the API easy for users (just implement DoFn and implement required SDF method if needed) and make validation the hard (slightly) way.